Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -3374,6 +3373,7 @@ void RenderLayer::paintLayerContents(GraphicsContext& context, const LayerPainti
     GraphicsContextStateSaver stateSaver(context, false);
     RegionContextStateSaver regionContextStateSaver(paintingInfo.regionContext);
 
+    localPaintFlags.remove(PaintLayerFlag::PaintingSVGClippingMask);
```

Comment:
I'd suggest to directly remove the PaintingSVGClippingMask from 'localPaintFlags' after we called setupClipPath() - this guarantees that we have no side effects and never propagate the PaintiningSVGClippingMask flags down the tree.

Reply:
Something like this:

```
diff --git a/Source/WebCore/rendering/RenderLayer.cpp b/Source/WebCore/rendering/RenderLayer.cpp
index 96f6d76acafb..4244e181b157 100644
--- a/Source/WebCore/rendering/RenderLayer.cpp
+++ b/Source/WebCore/rendering/RenderLayer.cpp
@@ -3373,10 +3373,13 @@ void RenderLayer::paintLayerContents(GraphicsContext& context, const LayerPainti
     GraphicsContextStateSaver stateSaver(context, false);
     RegionContextStateSaver regionContextStateSaver(paintingInfo.regionContext);
 
-    localPaintFlags.remove(PaintLayerFlag::PaintingSVGClippingMask);
     if (shouldApplyClipPath(paintingInfo.paintBehavior, localPaintFlags))
         setupClipPath(context, stateSaver, regionContextStateSaver, paintingInfo, localPaintFlags, columnAwareOffsetFromRoot);
 
+    bool applySVGClippingMask = localPaintFlags.contains(PaintLayerFlag::PaintingSVGClippingMask);
+    if (applySVGClippingMask)
+        localPaintFlags.remove(PaintLayerFlag::PaintingSVGClippingMask);
+
     bool selectionAndBackgroundsOnly = paintingInfo.paintBehavior.contains(PaintBehavior::SelectionAndBackgroundsOnly);
     bool selectionOnly = paintingInfo.paintBehavior.contains(PaintBehavior::SelectionOnly);
 
@@ -3512,7 +3515,7 @@ void RenderLayer::paintLayerContents(GraphicsContext& context, const LayerPainti
             paintMaskForFragments(layerFragments, context, paintingInfo, paintBehavior, subtreePaintRootForRenderer);
         }
 
-        if (localPaintFlags & PaintLayerFlag::PaintingSVGClippingMask || (!(paintFlags & PaintLayerFlag::PaintingCompositingMaskPhase) && (paintFlags & PaintLayerFlag::PaintingCompositingClipPathPhase))) {
+        if (applySVGClippingMask || (!(paintFlags & PaintLayerFlag::PaintingCompositingMaskPhase) && (paintFlags & PaintLayerFlag::PaintingCompositingClipPathPhase))) {
             // Re-use paintChildClippingMaskForFragments to paint black for the compositing clipping mask.
             paintChildClippingMaskForFragments(layerFragments, context, paintingInfo, paintBehavior, subtreePaintRootForRenderer);
         }
```